### ISSUE TITLE
exception while parsing first 100 bitcoin blk files

### ISSUE THREAD

**Issue by gimlids** [alecalve/python-bitcoin-blockchain-parser#47]
Is this something I should just catch and ignore? I'm doing aggregate statistics on the whole ledger, so I don't need absolute accuracy for every single transaction.

```
  File "/home/205740/.local/lib/python3.5/site-packages/blockchain_parser/output.py", line 60, in addresses
    if self.type == "pubkey":
  File "/home/205740/.local/lib/python3.5/site-packages/blockchain_parser/output.py", line 101, in type
    if self.is_pubkey():
  File "/home/205740/.local/lib/python3.5/site-packages/blockchain_parser/output.py", line 84, in is_pubkey
    return self.script.is_pubkey()
  File "/home/205740/.local/lib/python3.5/site-packages/blockchain_parser/script.py", line 103, in is_pubkey
    return len(self.operations) == 2 \
  File "/home/205740/.local/lib/python3.5/site-packages/blockchain_parser/script.py", line 74, in operations
    self._operations = list(self.script)
  File "/home/205740/.local/lib/python3.5/site-packages/bitcoin/core/script.py", line 622, in __iter__
    for (opcode, data, sop_idx) in self.raw_iter():
  File "/home/205740/.local/lib/python3.5/site-packages/bitcoin/core/script.py", line 607, in raw_iter
    raise CScriptTruncatedPushDataError('%s: truncated data' % pushdata_type, data)
bitcoin.core.script.CScriptTruncatedPushDataError: PUSHDATA(1): truncated data
```

**Comment by alecalve** [comment]
Ah you should probably ignore it, some transactions have invalid scripts. Which transaction is causing this?

**Comment by epicsol-inc** [comment]
Facing same problem, any ideas how can i fix it or if there is been any new commits for the fix?

**Comment by alecalve** [comment]
Do you know which block/transaction is causing this?

**Comment by epicsol-inc** [comment]
File: blk00088.dat
Block(000000000000000b7e48f88e86ceee3e97b4df7c139f5411d14735c1b3c36791)
Transaction(ebc9fa1196a59e192352d76c0f6e73167046b9d37b8302b6bb6968dfd279b767)

**Comment by vincentbdb** [comment]
same error and same Block for me, any update?

**Comment by shayanb** [comment]
The issue is most probably with parsing OP_return outputs, the transaction mentioned has 8 outputs all OP_returns.

**Comment by alecalve** [comment]
Should be fixed now that #11 is fixed
